### PR TITLE
Introduce api gateway lambda proxy resource

### PR DESCRIPTION
**Pull request by JakeGinnivan**
We have a common setup in Serverless Mono and Auth that we are using more often. 

## Why

There are two main approaches you can take with API Gateway and Lambda

1) Lambda per route, and routing done by API gateway
2) Single lambda which uses something like https://github.com/fastify/aws-lambda-fastify or an similar express library to do the routing in your app and API gateway just proxy all requests to a single lambda.

This makes it easy to setup the second

### 1. Local dev

There isn't a good local dev story for approach number 1, the best way is to push your pulumi stack up which is a slow cycle time.
In The Game we created an express server for local dev. When you add a new route you need to create a handler, then a lambda function, an API gateway route and an express handler which is registered with express.

This approach means add the route to fastify/express and it will work for both API gateway+lambda and local dev.

### 2. Deployment speeds

Deploying a single lambda plus it's associated resources is much faster than heaps of lambdas

### 3. Easier to monitor / navigate

When you split up into a bunch of lambdas, they need to be monitored separately. The best approach is with dashboards, which we haven't setup yet.

## Downsides

One of the reasons we didn't take this approach to start with is there are security downsides. With separate lambdas with their own policies you can restrict down to what each lambda needs.

It also makes the lambda smaller because it is doing less, making execution time slightly better.

That said, when comparing to our current ECS (containers) setup, we are not losing anything, but we are gaining the flexibility and cheap run cost of lambda

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
Need to merge this to master for the comment triggers to work :(

File: .github/workflows/pr-test-on-comment.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  issue_comment:
+    types: [created]
+
+jobs:
+  slashCommandDispatch:
+    runs-on: ubuntu-latest
+    if: github.event.review.state == 'approved'
+    steps:
+      - name: Slash Command Dispatch
+        uses: peter-evans/slash-command-dispatch@v2
```
Comment (JakeGinnivan) [inline .github/workflows/pr-test-on-comment.yml:13]: This command will make it easier to improve the /deploy command in news mono too!

**Comment by unique3** [comment]
how does restricting to write permission mitigate a malicious commit exposure?

File: libs/apigateway-lambda-proxy/src/lib/apigateway-lambda-proxy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        pulumiDebug('Stack created: %O', {
+            summary: stackUpResult.summary,
+            outputs: stackUpResult.outputs,
+        })
+    } catch (err) {
+        console.error({ err }, 'Stack creation failed')
+    }
+})
+
+it('can deploy lambda proxy', async () => {
```
Comment (unique3) [inline libs/apigateway-lambda-proxy/src/lib/apigateway-lambda-proxy.spec.ts:102]: nice, a second lambda to prove routing would be reassuring too

Comment (JakeGinnivan) [reply libs/apigateway-lambda-proxy/src/lib/apigateway-lambda-proxy.spec.ts:102]: This resource is only designed to have a single lambda, then use aws-fastify-lambda to map the API gateway route into a fastify request. 

The test is ensuring the proxy configuration is working and the full path comes through, which it does.

**Comment by JakeGinnivan** [comment]
> how does restricting to write permission mitigate a malicious commit exposure?

Only people with write permissions can review, then https://github.com/peter-evans/slash-command-dispatch by default will restrict the dispatch to anyone with at least write permission.

So if the pull request contributor doesn't have write access to the repo, it won't trigger anything.

**Comment by cyruseftos** [comment]
Thoughts about adding optional authorizer?

**Review by cyruseftos — APPROVED** [review]

File: jest.preset.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
-const nxPreset = require('@nrwl/jest/preset')
-
-module.exports = { ...nxPreset }
+module.exports = {
```
[the remark this answers is not in the document]
Comment (JakeGinnivan) [reply jest.preset.js:1]: Unsure, I am currently writing a blog post to document the switch to TypeScript project references

**Comment by JakeGinnivan** [comment]
> Thoughts about adding optional authorizer?

I was thinking we would do that inside fastify going forward

**Pull request merged**